upload/helpers.py

```diff
@@ -649,7 +649,20 @@ def dispatch_upload_task(task_arguments, repository, redis):
         immutable=True,
     )
 
-    notified = CommitNotification.objects.filter(commit__commitid=commitid).exists()
+    # we have a CommitNotifications model for recording if a notification
+    # has been sent for a particular commit but we'd like this early notification
+    # (below) to only happen once for a PR.  Checking the `commentid` isn't strictly
+    # correct but works for now since the early notify is only for PR comments.
+
+    notified = False
+    commit = Commit.objects.filter(commitid=commitid).first()
+    if commit and commit.pullid is not None:
+        pull = Pull.objects.filter(
+            repository_id=commit.repository_id, pullid=commit.pullid
+        ).first()
+        if pull and pull.commentid:
+            notified = True
```

Comment:
@dana-yaish What do you think about this approach?  I had been checking `CommitNotification` previously but that resulted in us showing the "processing" comment for a brief period each time a new commit was pushed to a PR.

Reply:
Isn't that more accurate though? Every time I push a new commit in a PR I expect the CI to run again and update me of the status of the new commit.

If we had 2 slightly different versions of "we are processing the upload" and "you updated this PR and we are processing the upload" that seems the most accurate to me.
It somehow seems better than the comment with the red banner saying "the head of this PR doesn't match these calculations, please upload new coverage"